authnz/src/main/java/io/stargate/auth/AuthorizationService.java

```diff
@@ -66,20 +71,16 @@ void authorizeDataWrite(String token, String keyspace, String table, Scope scope
       throws UnauthorizedException;
 
   /**
-   * Using the provided token will perform pre-authorization where possible and if successful
-   * executes the query provided.
+   * Using the provided token will perform pre-authorization where possible.
    *
-   * @param action A {@link QueryBuilder} object to be executed and authorized against a token.
    * @param token The authenticated token to use for authorization.
    * @param typedKeyValues A list of {@link TypedKeyValue} that will be used in the query and should
    *     be authorized against the token.
-   * @return On success will return the result of the query and otherwise will return an exception
-   *     relating to the failure to authorize.
-   * @throws Exception An exception relating to the failure to authorize.
+   * @param scope The {@link Scope} of the action to be performed.
+   * @throws UnauthorizedException An exception relating to the failure to authorize.
    */
-  ResultSet authorizedDataWrite(
-      Callable<ResultSet> action, String token, List<TypedKeyValue> typedKeyValues)
-      throws Exception;
+  void authorizedDataWrite(String token, List<TypedKeyValue> typedKeyValues, Scope scope)
+      throws UnauthorizedException;
```

Comment:
nit: Since we not longer returning the `ResultSet`, shall we rename to `authorizeDataWrite` (drop the `d`) following the pattern with `authorizeSchemaRead`, etc?

Reply:
Sure, will change after I merge into the base branch to avoid waiting on CI.